preprocessors/object-detection-azure/objdetect.py

```diff
@@ -0,0 +1,190 @@
+                    result = response.json()
+                    label = process_results(result)
+                else:
+                    return "", 204
+            else:
+                return "", 204
+
+    else:
+        return "", 204
+
+    return label
+
+
+
+@app.route("/preprocessor", methods=['POST', ])
+def categorise():
+    logging.debug("Received request")
+    # load the schema
+    with open('./schemas/preprocessors/object-detection.schema.json') \
+            as jsonfile:
+        data_schema = json.load(jsonfile)
+    with open('./schemas/preprocessor-response.schema.json') \
+            as jsonfile:
+        schema = json.load(jsonfile)
+    with open('./schemas/definitions.json') as jsonfile:
+        definitionSchema = json.load(jsonfile)
+    with open('./schemas/request.schema.json') as jsonfile:
+        first_schema = json.load(jsonfile)
+    schema_store = {
+        schema['$id']: schema,
+        definitionSchema['$id']: definitionSchema
+    }
+    resolver = jsonschema.RefResolver.from_schema(
+        schema, store=schema_store)
+
+    content = request.get_json()
+    try:
+        validator = jsonschema.Draft7Validator(first_schema, resolver=resolver)
+        validator.validate(content)
+    except jsonschema.exceptions.ValidationError as e:
+        logging.error(e)
+        return jsonify("Invalid Preprocessor JSON format"), 400
+    request_uuid = content["request_uuid"]
+    timestamp = time.time()
+    name = "ca.mcgill.a11y.image.preprocessor.objectDetectionAzure"
+    preprocess_output = content["preprocessors"]
+    classifier_1 = "ca.mcgill.a11y.image.preprocessor.contentCategoriser"
+    # convert the uri to processable image
+    if "graphic" not in content.keys():
+        return "", 204
+    else:
+        if classifier_1 in preprocess_output:
+            classifier_1_output = \
+                preprocess_output[classifier_1]
+            classifier_1_label = \
+                classifier_1_output["category"]
+            if classifier_1_label == "photograph":
+                source = content["graphic"]
+                image_b64 = source.split(",")[1]
+                binary = base64.b64decode(image_b64)
+                pred = process_image(image=binary)
+                type = {"objects": pred}
+            else:
+                """If the first classifier does not detect a photograph
+                the second classifier should not process the request"""
+                return "", 204
+        else:
+            """We are providing the user the ability to process a photograph
+            even when the first classifier is absent, however it is
+            recommended that the second classifier be used in conjunction
+            with the first classifier."""
+            source = content["graphic"]
+            image_b64 = source.split(",")[1]
+            binary = base64.b64decode(image_b64)
+            pred = process_image(image=binary)
+            type = {"objects": pred}
```

Comment:
Lots of labels have varying levels of specificity ("plant" vs "house plant" being returned in the same image). Do we get any more information back from azure?

Reply:
Also some are capitalized and others aren't ("chair" vs "Desk").

Reply:
I don' think we can get that information from azure.